metricbeat/docs/developer-guide/metricset-details.asciidoc

```diff
@@ -56,57 +56,56 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 [float]
 === Timeout Connections to Services
 
-As each time the Fetch method is called, it makes a request to the service, it
-is important to handle the connections correctly. It is recommended to setup the
-connections in the New method and persist them in the MetricSet object. This allows
+Each time the `Fetch` method is called, it makes a request to the service, so it's
+important to handle the connections correctly. We recommended that you set up the
+connections in the `New` method and persist them in the `MetricSet` object. This allows
 connections to be reused.
 
 One very important point is that connections must respect the timeout variable:
-`base.Module().Config().Timeout`. After the timeout a request must be ended and an error must
-be returned to make sure the next request can be started on time. By default the
-Timeout is set to Period, so one request gets ended when a new request is made.
+`base.Module().Config().Timeout`. If the timeout elapses before the request completes,
+the request must be ended and an error must be returned to make sure the next request
+can be started on time. By default the Timeout is set to Period, so one request gets
+ended when a new request is made.
 
-In case of a request must be ended or has an error, it should be make sure to
-return a good error message. This error message is also sent to Elasticesarch in
-index. This makes it possible to not only fetch metrics from the service, but
-also report potential problems / errors with the metricset.
+If a request must be ended or has an error, make sure that you return a good error
```

Comment:
Changed this to say, "If a request must be ended or has an error, make sure that you return an error message that contains enough detail for the user to diagnose the problem." We're kind of stating the obvious, but telling people to write  good/readable/understandable messages seems a tad vague. If we're going to say anything, we should qualify what we mean.